mmcv/ops/csrc/common/cuda/border_align_cuda_kernel.cuh

```diff
@@ -186,14 +186,16 @@ __global__ void border_align_backward_cuda_kernel(
                                   x_high, y_low, y_high, index);
 
     // update grad_output
-    atomicAdd(offset_grad_input + y_low * width + x_low,
-              *offset_grad_output * w1);
-    atomicAdd(offset_grad_input + y_low * width + x_high,
-              *offset_grad_output * w2);
-    atomicAdd(offset_grad_input + y_high * width + x_low,
-              *offset_grad_output * w3);
-    atomicAdd(offset_grad_input + y_high * width + x_high,
-              *offset_grad_output * w4);
+    if (x_low >= 0 && x_high >= 0 && y_low >= 0 && y_high >= 0) {
```

Comment:
Seems that if the coordinate is illegal, the gradient will not be propagated. Is this desirable? Should we clamp the gradient?

Reply:
The invalid coordinate contribute nothing when forward.

https://github.com/open-mmlab/mmcv/blob/304f184c461140c655d1c0b45fc6320e7bbdae87/mmcv/ops/csrc/common/cuda/common_cuda_helper.hpp#L33

https://github.com/open-mmlab/mmcv/blob/304f184c461140c655d1c0b45fc6320e7bbdae87/mmcv/ops/csrc/common/cuda/common_cuda_helper.hpp#L78-L83

So I think we do not need to propagate gradient on it.
And most `roi-align` like ops apply the same strategy, for example:
https://github.com/open-mmlab/mmcv/blob/304f184c461140c655d1c0b45fc6320e7bbdae87/mmcv/ops/csrc/common/cuda/deform_roi_pool_cuda_kernel.cuh#L153
https://github.com/open-mmlab/mmcv/blob/304f184c461140c655d1c0b45fc6320e7bbdae87/mmcv/ops/csrc/common/cuda/roi_align_cuda_kernel.cuh#L140
https://github.com/open-mmlab/mmcv/blob/304f184c461140c655d1c0b45fc6320e7bbdae87/mmcv/ops/csrc/common/cuda/rotated_feature_align_cuda_kernel.cuh#L120